cmd/systray/systray.go

```diff
@@ -337,14 +339,27 @@ func (menu *Menu) eventLoop(ctx context.Context) {
 					log.Printf("failed setting exit node: %v", err)
 				}
 			}
-			menu.rebuild(fetchState(ctx))
 
 		case <-menu.quit.ClickedCh:
 			systray.Quit()
 		}
 	}
 }
 
+// onClick registers a click handler for a menu item.
+func onClick(ctx context.Context, item *systray.MenuItem, fn func()) {
```

Comment:
should fn take the context too? you have it.

Then the funcs can select on ctx.Done themselves and not block forever on shutdown

Reply:
done